### PR TITLE
disable gene editing for other biomarkers, and 2 bug fixes

### PR DESCRIPTION
**Pull request by JiaoJiao123**
1) Releated to Issue #651
2) Bug fixes:
a) clear lastReviewed content for rollback when they enter the review mode, instead of the moment that rollback happens.
b) add userRole check for display obsolete

### CODE REVIEW

File: web/yo/app/scripts/controllers/gene.js [hunk 1]
```diff
@@ -3589,6 +3599,7 @@ angular.module('oncokbApp')
                         'Sorry for any inconvinience.');
                 }
                 $scope.fileEditable = false;
+                $scope.geneEditable = $scope.fileEditable;
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: Shouldn't you just make a watcher on fileEditable?

File: web/yo/app/scripts/directives/estatus.js [hunk 2]
```diff
@@ -18,11 +18,6 @@ angular.module('oncokbApp')
             },
             replace: true,
             link: function postLink(scope, element) {
-                $rootScope.$watch('reviewMode', function(n, o) {
```
Comment (JiaoJiao123) [inline web/yo/app/scripts/directives/estatus.js]: not in eStatus.js. It was removed in review panel directive js.

File: web/yo/app/scripts/controllers/gene.js (hunk 1 above)
Comment (JiaoJiao123) [inline web/yo/app/scripts/controllers/gene.js]: I was thinking about that. Should I? Since we have been trying to reduce watcher usage.

Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: We should. Having a watcher is not a bad practice. It's actually designed to deal with the situation like this.

File: web/yo/app/scripts/controllers/gene.js [hunk 3]
```diff
@@ -2885,6 +2891,7 @@ angular.module('oncokbApp')
                     // This will only happen if the currentReviewer is not empty
                     $scope.fileEditable = false;
                 }
+                $scope.geneEditable = $scope.fileEditable;
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: This can be removed?

Comment (JiaoJiao123) [inline web/yo/app/scripts/controllers/gene.js]: I don't think so. This is the first initialization.

Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: At line 2886, when the moment you change fileEditable, shouldn't it trigger the watcher to update geneEditable already?

Comment (JiaoJiao123) [inline web/yo/app/scripts/controllers/gene.js]: What you said will happen only in some cases. 
When file.editable is false, the watch on fileEditable is not going to be triggered. But at this time, the condition check in displayCheck() will be executed, and geneEditable will be assigned to be false. So I removed it

**Pull request merged**